covigator/references/lineage_annotation.py

```diff
@@ -131,7 +579,68 @@ def _fill_lineage_table(self, lineage_constellation):
                 self.session.add(lineage)
                 self.session.commit()
                 count_lineages += 1
+            # Append all lineage defining mutations to a list of seen mutations
+            for this_mut in annotation["lineage_mutations"]:
+                # Skip variants on nucleotide level
+                if this_mut["type"] == "INSERTION":
+                    continue
+                if this_mut["variant_id"] not in seen_mutations:
+                    seen_mutations.add(this_mut["variant_id"])
+                    all_mutations.append(this_mut)
         logger.info("Loaded into the database {} lineages".format(count_lineages))
+        
+        # Store all observed lineage defining mutations in database
+        for this_mut in all_mutations:
+            mutation = LineageDefiningVariants(
+                variant_id=this_mut["variant_id"],
+                hgvs=this_mut["hgvs_p"],
+                variant_type=this_mut["type"],
+                protein=this_mut["protein"],
+                position=this_mut["position"],
+                reference=this_mut["reference"],
+                alternate=this_mut["alternate"],
+                ambiguous_alternate=this_mut["ambiguous_alternate"],
+                annotation=this_mut["annotation"],
+                variant_level=this_mut["level"]
+            )
+            self.session.add(mutation)
+            self.session.commit()
```

Comment:
You could do a single commit out of the loop or even better you can use the `self.session.add_all()` method. Taking this to the extreme this can be a performance issue.